pkg/arvo/app/btc-node-hook.hoon

```diff
@@ -0,0 +1,354 @@
+::  btc-node-hook: send JSON rpc requests to bitcoin full node
+::  and poke the responses into the btc-node-store
+::
+/-  *btc-node-hook, *btc-node-store, sole
+/+  default-agent, sole, lib=btc-node-json, verb
+::
+=>  |%
+    +$  card  card:agent:gall
+    +$  versioned-state
+      $%  [%0 state-zero]
+      ==
+    ::
+    +$  state-zero
+      $:  endpoint=@t
+          headers=header-list:http
+      ==
+    --
+::
+=|  state-zero
+=*  state  -
+::  Main
+::
+%+  verb  |
+^-  agent:gall
+=<  |_  =bowl:gall
+    +*  this      .
+        btc-core  +>
+        bc        ~(. btc-core bowl)
+        def       ~(. (default-agent this %|) bowl)
+    ::
+    ++  on-init
+      ^-  (quip card _this)
+      :-  ~
+      %=  this
+        endpoint  ''
+        headers   ~
+      ==
+    ::
+    ++  on-save   !>(state)
+    ++  on-load
+      |=  old=vase
+      `this(state !<(state-zero old))
+    ::
+    ++  on-poke
+      |=  [=mark =vase]
+      ^-  (quip card _this)
+      |^
+      ?+    mark    (on-poke:def mark vase)
```

Comment:
imo it looks better to indent the "main" block of the `|^`, either one level, or putting is directly after the `|^` like you would with `:~`.

You might also consider moving the `+hook-action` and `+hook-command` into the `+bc` below, to keep the gall interface core here as simple as possible. Up to you though. Again, no well-established standards here yet.

Reply:
First suggestion, done.
Second suggestion... I might leave it as it, for the time being (I'd like to get this out of the way asap (: ) But I agree with you that keeping the gall core as spartan as possible is desirable.